### PR TITLE
switch clientset test to use a configmap

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/kubernetes/kubernetes/issues/42724

### CODE REVIEW

**Comment by deads2k** [comment]
Pods are a really contentious resource with lots of controllers messing with them.  If we just want to make sure the clientset works, we can use a configmap instead.

File: test/e2e/generated_clientset.go [hunk 1]
```diff
@@ -100,12 +100,19 @@ func observeCreation(w watch.Interface) {
 }
 
 func observeObjectDeletion(w watch.Interface) (obj runtime.Object) {
+	// output to give us a duration to failure.  Maybe we aren't getting the
+	// full timeout for some reason.  My guess would be watch failure
+	framework.Logf("Starting to observe pod deletion")
```
Comment (marun) [inline test/e2e/generated_clientset.go]: pod -> configmap?

**Comment by marun** [comment]
/lgtm

I think the longer-term solution (as discussed with @ncdc and @caesarxuchao  on slack) is doing this in an integration test, but switching to a resource that doesn't involve anything but creation/deletion seems like a reasonable stop-gap.

File: test/e2e/generated_clientset.go [hunk 2]
```diff
@@ -122,71 +129,73 @@ func observeObjectDeletion(w watch.Interface) (obj runtime.Object) {
...
-		By("verifying pod creation was observed")
+		By("verifying configmap creation was observed")
 		observeCreation(w)
 
-		// We need to wait for the pod to be scheduled, otherwise the deletion
-		// will be carried out immediately rather than gracefully.
-		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
-
-		By("deleting the pod gracefully")
-		if err := podClient.Delete(pod.Name, metav1.NewDeleteOptions(30)); err != nil {
```
Comment (ncdc) [inline test/e2e/generated_clientset.go]: I believe @caesarxuchao wanted to test that graceful deletion was set correctly on the pod

Comment (deads2k) [inline test/e2e/generated_clientset.go]: > I believe @caesarxuchao wanted to test that graceful deletion was set correctly on the pod

A worth goal, but seems like that would be a very different test which didn't need to watch for a deletion.  I'd say a similar thing for the bug this is exposing.  There should be test focused on that in particular, but that the goal of having a 1.5 client work doesn't require it.

**Comment by caesarxuchao** [comment]
Can we leave a test that exercises the deleteOptions? DeleteOptions is encoded into body, so it's quite special and i want a test to cover it. I suggest that we remove the "watch for deletion" part of the original test, instead verifying that the Pod's DeletionTimeStamp becomes non-nil.

**Comment by deads2k** [comment]
> Can we leave a test that exercises the deleteOptions? DeleteOptions is encoded into body, so it's quite special and i want a test to cover it. I suggest that we remove the "watch for deletion" part of the original test, instead verifying that the Pod's DeletionTimeStamp becomes non-nil.

Sure.  Tag me when you have a pull ready for review.

**Pull request closed without merging**